app/models/host.rb

```diff
@@ -30,6 +35,14 @@ class Host < ApplicationRecord
   has_many :assigned_internal_profiles, -> { external(false) },
            through: :policies, source: :profiles
 
+  scope :with_benchmark, lambda { |profile = nil|
+    profile ||= RequestStore.store['scoped_search_context_profiles']
+
+    left_outer_joins(test_result_profiles: :benchmark).where(
```

Comment:
@vkrizan something like this?

Reply:
Yes. It just needs to join with exactly one test result, benchmark and profile: with latest test result for the given profile scope.

The `scoped_search_context_profiles` would likely be the initial profile or any profile for a given policy, wouldn't it?

Reply:
It's all the child profiles under the policy

Reply:
The "It just needs to join with exactly one test result, benchmark and profile: with latest test result for the given profile scope." is not resolved, yet.

Reply:
@vkrizan after our discussion with @LightOfHeaven1994 are you okay with this?

Reply:
Yes, since the system keeps one test result for each policy for a host, this implementation make sense.